### ISSUE TITLE
[suggestion] Show flattened subfolders in folder pane

### ISSUE THREAD

**Issue by lofidevops** [kmwallio/ThiefMD#182]
**Problem statement:** Navigating nested folders prevents seeing all files at once.

**Suggested solution:**

* Library pane shows only top-level folder (root folder).
* Folder pane shows: files in top-level folder, header for each subfolder, under each header, the files in that subfolder.
* Edit pane remains the same.

This would be useful, for example, opening a typical code repository with a `README.md` in the root folder, and a bunch of Markdown files in the `docs` subfolder. All Markdown files would be visible without navigation. I *imagine* it would be useful for someone writing a book using subfolders, but I may be mistaken.

If the motivation or suggestion is unclear, let me know and I will draft a more detailed example. If it doesn't fit in the roadmap, oh well.

**Comment by kmwallio** [comment]
Is this like if I had:

```
/Novel
/Novel/title-page.md
/Novel/Chapter-01/Part-1.md
/Novel/Chapter-01/Part-2.md
/Novel/Chapter-01/Part-3.md
/Novel/Chapter-02/Part-X.md
/Novel/Chapter-0Y/...
```

I'd only see "Novel" in the library, and the Folder pane would have:

```
title-page.md
[Chapter-01]
Part-1.md
Part-2.md
Part-3.md
[Chatper-01]
Etc...
```

I like the idea. We have an idea for Project Types. Novel's would work well for this view, but something like a static site might light the hierarchy view.

We can probably add a "Show Nested Folders" option for the Folder pane, but the Library pane I'll have to think about before committing too. Maybe it defaults to collapsed if Show Nested Folders is chosen.

**Comment by lofidevops** [comment]
Exactly. An option for the folder pane plus collapse-by-default sounds sensible to me.
